src/utils/attributes.js

```diff
@@ -0,0 +1,49 @@
+
+const CSSStyleMap = {
+    stroke: "stroke",
+    strokeWidth: "stroke-width",
+    strokeDasharray: "stroke-dasharray",
+    fill: "fill",
+    fillOpacity: "fill-opacity",
+    id: "id",
+    className: "class"
+}
+
+const defaultAttributes = {
+    stroke: "black",
+    strokeWidth: 1,
+    strokeDasharray: "",
+    fill: "none",
+    fillOpacity: 0.8
+}
+
+class SVGAttributes {
+    constructor(args = defaultAttributes) {
+        for(const property in args) {
+            this[property] = args[property]
+        }
+        this.stroke = args.stroke ?? defaultAttributes.stroke
+        this.strokeWidth = args.strokeWidth ?? defaultAttributes.strokeWidth
+        this.strokeDasharray = args.strokeDasharray ?? defaultAttributes.strokeDasharray
+        this.fill = args.fill ?? defaultAttributes.fill
+        this.fillOpacity = args.fillOpacity ?? defaultAttributes.fillOpacity
+    }
+
+    toJSON() {
+        let attrCSSStyle = Object.keys(this)
+            .reduce( (acc, key) =>
+                    acc + (this[key] ? `${CSSStyleMap[key]}="${this[key]}" ` : "")
```

Comment:
As mentioned in https://github.com/alexbol99/flatten-js/issues/130#issuecomment-1322278284, it would be more helpful to allow arbitrary `key`s, via `CSSStyleMap[key] ?? key`. Then you also wouldn't need to explicitly list `id` in `CSSStyleMap`. But perhaps you've already decided against that. If I can argue for it again, this would be helpful for `data-*` (custom) attributes.